### PR TITLE
Correctly loop over possible firefoxes

### PR DESCRIPTION
**Pull request by 6a68**
@ianb Think I've got it this time. Mind taking a look?

### CODE REVIEW

**Comment by 6a68** [comment]
Hmm, but this does not fix #4486

**Comment by 6a68** [comment]
Now fixes #4486 too

File: bin/run-addon [hunk 1]
```diff
@@ -9,20 +9,20 @@ jpm="$base/node_modules/.bin/jpm"
...
-$(which firefox || true)
-"
+firefoxes=(
+"/Applications/FirefoxNightly.app"
+"/Applications/Firefox Nightly.app"
+"/Applications/FirefoxDeveloperEdition.app"
+"/Applications/Firefox Developer Edition.app"
+"/Applications/FirefoxAurora.app"
+"$(which firefox || true)"
+)
```
Comment (chenba) [inline bin/run-addon]: It probably makes more sense to have `firefox` first?  That way I can change wherever `firefox` is symlink'd to run a different version of Firefox.

**Comment by 6a68** [comment]
@chenba updated. r?

File: bin/run-addon [hunk 2]
```diff
@@ -94,7 +94,7 @@ done
 open=$binary # assumes Linux
 unamestr=$(uname)
 if [[ "$unamestr" = "Darwin" ]]; then
-  open="open $binary"
+  open=(open "${binary}")
```
Comment (chenba) [inline bin/run-addon]: I don't quite understand why this needs to be an array.  Otherwise LGTM.

Comment (6a68) [reply bin/run-addon]: I'm not much of a bash scripter. Suggestions welcome :-)

Comment (chenba) [reply bin/run-addon]: The older code of having `open` as a string should work as far as I can tell.

Comment (chenba) [reply bin/run-addon]: Oh, the spaces.  :satisfied:

**Pull request merged**